docs/reference/asyncapi-rules.md

```diff
@@ -0,0 +1,368 @@
+  description: >+
+    Some description about the general point of this API, and why it exists when another similar but different API also exists.
+
+    ## Authentication
+
+    This API uses OAuth2 and tokens can be requested from [Dev Portal: Tokens](https://example.org/developers/tokens).
+```
+
+### asyncapi-info-license-url
+
+Mentioning a license is only useful if people know what the license means, so add a link to the full text for those who need it.
+
+**Recommended:** No
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+    url: https://www.tldrlegal.com/l/mit
+```
+
+### asyncapi-info-license
+
+The `info` object should have a `license` key.
+
+It can be hard to pick a license, so if you don't have a lawyer around you can use [TLDRLegal](https://tldrlegal.com/) and [Choose a License](https://choosealicense.com/) to help give you an idea.
+
+How useful this is in court is not entirely known, but having a license is better than not having a license.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+```
+
+### asyncapi-operation-description
+
+Operation objects should have a description.
+
+**Recommended:** Yes
+
+### asyncapi-operation-operationId
+
+This operation ID is essentially a reference for the operation. Tools may use it for defining function names, class method names, and even URL hashes in documentation systems.
+
+**Recommended:** Yes
+
+### asyncapi-parameter-description
+
+Parameter objects should have a `description`.
+
+**Recommended:** No
+
+### asyncapi-payload-default
+
+`default` objects should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+**Good Example**
+
+``` yaml
+payload:
+  type: object
+  properties:
+    value:
+      type: integer
+  required:
+    - value
+  default:
+    value: 17
+```
+
+**Bad Example**
+
+``` yaml
+payload:
+  type: object
+  properties:
+    value:
+      type: integer
+  required:
+    - value
+  default:
+    value: nope!
+```
+
+### asyncapi-payload-examples
+
+Values of the `examples` array should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+**Good Example**
+
+``` yaml
+payload:
+  type: object
+  properties:
+    value:
+      type: integer
+  required:
+    - value
+  examples:
+    - value: 13
+    - value: 17
+```
+
+**Bad Example**
+
+``` yaml
+payload:
+  type: object
+  properties:
+    value:
+      type: integer
+  required:
+    - value
+  examples:
+    - value: nope!       # Wrong type
+    - notGoodEither: 17  # Missing required property
+```
+
+### asyncapi-payload-unsupported-schemaFormat
+
+Spectral currently only supports payload validation against the default AsyncAPI schema.
+
+This rule will notify you when a payload is decorated with a `schemaFormat` property. Spectral won't be able to validate it and potential issues in this payload will go undetected by the `asyncapi-payload-*` rules.
+
```

Comment:
```suggestion
AsyncAPI can support various `schemaFormat` values, with the default being one of the following:

```
application/vnd.aai.asyncapi;version=2.0.0
application/vnd.aai.asyncapi+json;version=2.0.0
application/vnd.aai.asyncapi+yaml;version=2.0.0
```

At this point no other schemaFormat's are supported by Spectral, so if you use them this rule will emit an `info` message and skip validating the payload. 

Other formats such as OpenAPI Schema Object, JSON Schema Draft 07 and Avro will be added in various upcoming versions. 
```

Reply:
@philsturgeon Awesome! I've slightly reworded your proposal (as we don't support setting the `schemaFormat` at all).